test.sh

```diff
@@ -52,6 +66,13 @@ for dir in ${VERSIONS}; do
   fi
 
   if [ -n "${TEST_OPENSHIFT_4}" ]; then
+    # In case only imagestream is deprecated
+    # and the other tests should be working
+    if [ -e ".exclude-openshift" ]; then
+      echo "-> .exclude-openshift file exists for version $dir, skipping OpenShift-4 tests."
+      clean_image
+      continue
+    fi
```

Comment:
I would rather see this being tested based on the current versions located inside the imagestreams rather than have a hidden file inside the versioned directories. Seems error-prone (if we forget to remove it for some future version).

Reply:
You are right, but what about this case. We remove the stream NodeJS-10 two months ago before NodeJS-10 is EOL.
Like we are planning.
As soon as I would like to test NodeJS-10 in OpenShift 3 or 4 then the imagestream does not exist, because it is not specified in imagestream. But a template requires imagestream for version 10.
The OpenShift tests will fail.

Reply:
Imo checking that the imagestream version is not present anymore and skipping the tests is a valid strategy here - the imagestream is EOL, it makes sense to not test it anymore.